treeherder/perfalert/perfalert/__init__.py

```diff
@@ -111,23 +108,15 @@ def __str__(self):
                                             self.machine_id)
 
 
-class TalosAnalyzer:
+class Analyzer:
     def __init__(self):
-        # List of PerfDatum instances
-        self.data = []
-        self.machine_history = {}
-
-    def addData(self, data):
-        self.data.extend(data)
-        for d in data:
-            self.machine_history.setdefault(d.machine_id, []).append(d)
+        self.data = []  # List of PerfDatum instances
 
+    def add_data(self, push_timestamp, value, **kwargs):
+        self.data.append(PerfDatum(push_timestamp, value, **kwargs))
```

Comment:
do we need to use PerfDatum here?  it seems as though we don't use it earlier on.